### PR TITLE
Fix perf on scrub

### PR DESCRIPTION
**Pull request by istarkov**
## Description

fixes perf on breakpoint scrub change

Still some issues on breakpoint change, but we know why this happens. Would be fixed in next iterations.

## Steps for reproduction

1. click button
2. expect xyz

## Code Review

- [ ] hi @kof, I need you to do
  - conceptual review (architecture, feature-correctness)
  - detailed review (read every line)
  - test it on preview

## Before requesting a review

- [ ] made a self-review
- [ ] added inline comments where things may be not obvious (the "why", not "what")

## Before merging

- [ ] tested locally and on preview environment (preview dev login: 5de6)
- [ ] updated [test cases](https://github.com/webstudio-is/webstudio-builder/blob/main/apps/builder/docs/test-cases.md) document
- [ ] added tests
- [ ] if any new env variables are added, added them to `.env.example` and the `builder/env-check.js` if mandatory

### CODE REVIEW

File: apps/builder/app/builder/features/style-panel/style-panel.tsx [hunk 1]
```diff
@@ -26,19 +27,37 @@ export const StylePanel = ({ selectedInstance, publish }: StylePanelProps) => {
...
+      }
+      if (canvasWidth === undefined) {
+        return;
+      }
+
+      if (matchMedia(breakpoint, canvasWidth) === false) {
+        setRenderAlert(true);
+      }
+
+      setRenderAlert(false);
```
Comment (kof) [inline apps/builder/app/builder/features/style-panel/style-panel.tsx:48]: did you forget a return?

File: apps/builder/app/builder/features/style-panel/style-panel.tsx [hunk 2]
```diff
@@ -26,19 +27,37 @@ export const StylePanel = ({ selectedInstance, publish }: StylePanelProps) => {
...
     selectedInstance === undefined ||
-    breakpoint === undefined ||
-    canvasWidth === undefined
+    breakpoint === undefined
   ) {
     return null;
   }
 
-  if (matchMedia(breakpoint, canvasWidth) === false) {
+  if (renderAlert) {
```
Comment (kof) [inline apps/builder/app/builder/features/style-panel/style-panel.tsx:60]: renderWarning

File: apps/builder/app/builder/features/style-panel/style-panel.tsx (hunk 1 above)
Comment (istarkov) [reply apps/builder/app/builder/features/style-panel/style-panel.tsx:48]: no its few lines above

File: apps/builder/app/builder/features/style-panel/style-panel.tsx [hunk 3]
```diff
@@ -26,19 +27,37 @@ export const StylePanel = ({ selectedInstance, publish }: StylePanelProps) => {
       publish,
     });
 
+  const [renderAlert, setRenderAlert] = useState(false);
+
   const breakpoint = useStore(selectedBreakpointStore);
-  const [canvasWidth] = useCanvasWidth();
+
+  useEffect(() => {
```
Comment (kof) [inline apps/builder/app/builder/features/style-panel/style-panel.tsx:34]: move it to a hook in the same file? useShowWarning

Comment (kof) [reply apps/builder/app/builder/features/style-panel/style-panel.tsx:34]: or useMatchMedia // true

**Review by kof — APPROVED** [review]

File: apps/builder/app/builder/features/style-panel/style-panel.tsx [hunk 4]
```diff
@@ -10,35 +10,60 @@ import type { Publish } from "~/shared/pubsub";
...
+      }
+      if (canvasWidth === undefined) {
+        return;
+      }
+
+      if (matchMedia(breakpoint, canvasWidth) === false) {
+        setMatched(true);
+      }
+
+      setMatched(false);
```
Comment (TrySound) [inline apps/builder/app/builder/features/style-panel/style-panel.tsx:40]: Do you intentionally set false right after true?

Comment (istarkov) [reply apps/builder/app/builder/features/style-panel/style-panel.tsx:40]: no

File: apps/builder/app/builder/features/style-panel/style-panel.tsx [hunk 5]
```diff
@@ -10,35 +10,60 @@ import type { Publish } from "~/shared/pubsub";
...
 import { StyleSourcesSection } from "./style-source-section";
 import { matchMedia } from "@webstudio-is/css-engine";
+import { useEffect, useState } from "react";
 
 type StylePanelProps = {
   publish: Publish;
   selectedInstance: Instance;
 };
 
+const useMatchMedia = () => {
```
Comment (TrySound) [inline apps/builder/app/builder/features/style-panel/style-panel.tsx:23]: Can be computed store.

```
computed([canvasWidthContainer, selectedBreakpointStore], (canvasWidth, selectedBreakpoint) => matchMedia(selectedBreakpoint, canvasWidth))
```

**Pull request merged**